diag_manager/fms_diag_object.F90

```diff
@@ -396,17 +422,323 @@ function diag_obj_is_static (obj) result (rslt)
     rslt = obj%static
 end function diag_obj_is_static
 
-function diag_obj_get_vartype (obj) result (rslt)
-    class(fmsDiagObject_type), intent(in) :: obj
-    integer :: rslt
-    rslt = obj%vartype
-end function diag_obj_get_vartype
-
-function diag_obj_get_varname(obj) result (rslt)
-    class(fmsDiagObject_type), intent(in) :: obj
-    character(len=len(obj%varname)) :: rslt
-    rslt = obj%varname
-end function diag_obj_get_varname
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!! Get functions
+
+!> @brief Gets metedata
+!! @return metadata string array, or a single space if metadata is not allocated
+function get_metadata (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+     character(len=:), allocatable, dimension(:) :: rslt
+     if (allocated(obj%metadata)) then
+       allocate(character(len=(len(obj%metadata(1)))) :: rslt (size(obj%metadata)) )
+       rslt = obj%metadata
+     else
+       allocate(character(len=1) :: rslt(1:1))
+       rslt = " "
+     endif
+end function get_metadata
+!> @brief Gets static
+!! @return true if the variable is static
+function get_static (obj) &
```

Comment:
What you have works, but we should discuss if its a good idea to  follow a convention that adds 
a function ( doesn;t get rig of the get_satic, but adds another ) named ``` is_static``` 
It is used like 
``` 
if (obj%is_static() ) then
```
In the end, waht you ahve passes the litmus test that these two statements  be
 equivalent (if  static had been public)
```
 s = obj%static
s = obj%get_static()
```
(this is because staic is another intrisic (logical) like int, real etc.